### PR TITLE
Rule: metasyntactic-variable

### LINKED ISSUE #244: Metasyntactic variable
https://en.wikipedia.org/wiki/Metasyntactic_variable

> Metasyntactic variables used commonly across all programming languages include [foobar](https://en.wikipedia.org/wiki/Foobar), foo, bar, baz, qux, quux, corge, grault, garply, waldo, fred, [plugh](https://en.wikipedia.org/wiki/Plugh), [xyzzy](https://en.wikipedia.org/wiki/Xyzzy_(computing)), and thud

This would be a good rule in the "testing" category (which is really more about "development" and should probably be renamed), as avoiding names like "foo" and "bar" seems like a reasonable thing to want in production policy.

### PR DESCRIPTION
**Pull request by anderseknert**
Add rule to check for "foo", "bar" and other metasyntactic variables.

Fixes #244

This PR also unifies the logic of this rule and prefer-snake-case, as well as their tests.

### CODE REVIEW

**Review by charlieegan3 — APPROVED** [review]
Looks good to be, never new foobar had a name!

File: bundle/regal/rules/style/prefer_snake_case.rego [hunk 1]
```diff
@@ -28,12 +25,9 @@ report contains violation if {
...
 }
 
-_is_name(ref, pos) if {
-	pos == 0
-	ref.type == "var"
-}
+# annoyingly, rule head refs are missing location when a rule.head.name is present,
+# or rather when there's only a single item in the ref.. this inconsistency should
+# probably be fixed in OPA, but until then, there's this.
+location_of(ref, rule) := ref if ref.location
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply bundle/regal/rules/style/prefer_snake_case.rego:31]: Yeah, I'll file an issue. Thanks for the reminder!

File: bundle/regal/rules/testing/metasyntactic_variable.rego [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import data.regal.ast
+import data.regal.result
+
+metasyntactic := {
+	"foobar",
+	"foo",
+	"bar",
+	"baz",
+	"qux",
+	"quux",
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply bundle/regal/rules/testing/metasyntactic_variable.rego:18]: No, me neither! I doubt there will be much overlap with legit variable names though 😆

[the remark this answers is not in the document]
Comment (charlieegan3) [reply bundle/regal/rules/testing/metasyntactic_variable.rego:18]: Yeah they seem pretty safe to me!

File: docs/rules/testing/metasyntactic-variable.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+- qux
+- quux
+- corge
+- grault
+- garply
+- waldo
+- fred
+- plugh
+- xyzzy
+- thud
```
[the remark this answers is not in the document]
Comment (Parsifal-M) [reply docs/rules/testing/metasyntactic-variable.md:47]: _**fred** will remember this..._ 😆

[the remark this answers is not in the document]
Comment (charlieegan3) [reply docs/rules/testing/metasyntactic-variable.md:47]: Ha yeah, I thought poor Fred too!

File: bundle/regal/rules/style/prefer_snake_case.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (anderseknert) [reply bundle/regal/rules/style/prefer_snake_case.rego:31]: Here goes: https://github.com/open-policy-agent/opa/issues/6199

**Pull request merged**